pkgs/misc/dxvk/darwin-dxvk-compat.patch

```diff
@@ -0,0 +1,35 @@
+Source: https://github.com/marzent/dxvk/commit/65247afe2de5dd6b555b8fdd46dddcdc6325e1d6
```

Comment:
What speaks against using their tree as the src on Darwin?

Having a patch that might have to be rebased in future releases wouldn't be great for Linux users who wouldn't even have the means to test in case of an update.

Reply:
> What speaks against using their tree as the src on Darwin?

The history in that repo is pretty messy, and it lags behind upstream sometimes. I had to go digging to find what was actually changed from upstream, so I could create the patch.

> Having a patch that might have to be rebased in future releases wouldn't be great for Linux users who wouldn't even have the means to test in case of an update.

I’m actively using DXVK on Darwin, so I don’t expect that to be an issue.

Reply:
> it lags behind upstream sometimes

That's kinda the point. If it lags behind upstream, the patch might not apply anymore if Linux users want to update DXVK.

Reply:
The patch isn’t used on Linux. Linux users wouldn’t be affected if someone bumped DXVK and Darwin didn’t build.

I’d also rather not use a different repo for Darwin when that repo makes use of merge commits to track upstream. I’ve had some bad experiences with merge commits at work and would rather maintain and apply the patches in the package instead.

Reply:
The problem is that if someone tried to bump DXVK on Linux, they wouldn't know what to do with the patch.

If we had two separate sources, each platform could be updated independently.

Reply:
I think it would be surprising if someone beat me to the punch, but that’s fair.

I’e made some changes that I’ll push when DXVK releases a new version.

* I separated the sync logic into its own patch and conditionally apply it. I tried to do this in a way that is robust to upstream changes.
* I retained one upstream source and the patch, but I set it up such that it should be easy to allow Darwin to diverge and use an older version if the patches don’t work on the latest one.

As for the MoltenVK issue, I’m still not really sure. The approach I’m considering is to `passthru` the patch and leave it to DXVK to apply it. It’s still not pretty, but it makes it clear to the user what they’re doing. No one should ever used the patched MoltenVK except with DXVK.

Reply:
It’s also possible the sync primitives patch performs better on Linux, but I lack x86_64-linux systems that are capable of playing games. Someone would need to test and determine whether that were the case (since nix uses a different threading implementation with MinGW that purports to be faster).

Reply:
I went ahead and opened #163474.